### PR TITLE
Use fixed Twig dep to fix NTS PHP build for Windows

### LINKED ISSUE #20709: Install fails with open_basedir and NTS PHP on Windows
related with previous issue https://github.com/PrestaShop/PrestaShop/issues/16736

## how to replicate
issue present when installing from CLI using NTS PHP with `open_basedir` set, steps to replicate:

1. download NTS PHP for Windows (ex https://windows.php.net/downloads/releases/php-7.4.9-nts-Win32-vc15-x64.zip )
2. clone the latest PrestaShop develop and run composer install
4. install from CLI using `php -d open_basedir="PrestaShop_directory" admin-dev/index_cli.php`

currently, the install does not succeed and it emits a php warning:
```
Warning: is_file(): open_basedir restriction in effect. File(C:\xxx\ps_php8\ps_17_php8\app/Resources/views/C:/xxx/ps_php8/ps_17_php8/mails/themes/modern/core/account.html.twig) is not within the allowed path(s): (C:\xxx\ps_php8)
```

some template path is built wrongly, ie. absolute file path is prefixed with a directory path

## local workaround

(to make install passing for development purposes)

delete `mails/themes/modern/` folder

### PR DESCRIPTION
**Pull request by mvorisek**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Use fixed Twig dep.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #20709, partly #22664
| How to test?      | CI must pass
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23896)

### CODE REVIEW

**Comment by mvorisek** [comment]
Finish after https://github.com/twigphp/Twig/pull/3499 is merged and released.

Please advise which branch to target.

File: composer.json
Comment (Progi1984) [inline composer.json]: Could you sync composer.lock for QA testing ?

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: composer.json
Comment (mvorisek) [reply composer.json]: No :), see desc, Twig not released yet...

Comment (Progi1984) [reply composer.json]: Ok. So we wait for release before review

File: composer.json
Comment (Progi1984) [inline composer.json:17]: Please revert this change

File: composer.json
Comment (Progi1984) [inline composer.json:146]: And this one

**Review by Progi1984 — COMMENTED** [review]
Please don't change PHP requirements

File: composer.json
Comment (mvorisek) [reply composer.json:17]: Improssible, created another PR for it https://github.com/PrestaShop/PrestaShop/pull/24476

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I see no reason to update dependencies for Windows right now, Syfmony 4.4 and PHP8 will update dependencies and maybe not this one.

**Comment by matks** [comment]
Hi everyone

I'm afraid here we are in a situation where we don't want to merge this PR. It has not been moved since May and I think if nobody from maintainer team joins in the discussion it means nobody from maintainer team is willing to join.

By opening this PR but it seems we don't want the project to go through this path in this direction, at least today. Maybe in the future.

Thank you for your contribution, I close this Pull Request because I don't think it will be merged.

If 2 other maintainers of the project disagree with me they are free to re-open the Pull Request and continue this journey. For me, it's better to halt it here.

**Comment by mvorisek** [comment]
reopen, this fixes a bug

**Pull request closed without merging**
